src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -102,8 +128,36 @@ BackupCoordinationStageSync::State BackupCoordinationStageSync::readCurrentState
         if (!zk_nodes_set.contains("current|" + host + "|" + stage_to_wait))
         {
             UnreadyHostState unready_host_state;
-            unready_host_state.started = zk_nodes_set.contains("started|" + host);
-            unready_host_state.alive = zk_nodes_set.contains("alive|" + host);
+            const String started_node_name = "started|" + host;
+            const String alive_node_name = "alive|" + host;
+            const String alive_node_path = zookeeper_path + "/" + alive_node_name;
+            unready_host_state.started = zk_nodes_set.contains(started_node_name);
+
+            /// Because we do retries everywhere we can't fully rely on ephemeral nodes anymore.
+            /// Though we recreate "alive" node when reconnecting it might be not enough and race condition is possible.
+            /// And everything we can do here - just retry.
+            unready_host_state.alive = zk_nodes_set.contains(alive_node_name);
```

Comment:
Where do we use `unready_host_state.alive` and what may happen after the race condition? This code does not protect from the race condition, so at least it would be great to know what will break and how bad are potential consequences